File: grazie-service/example_usage.py
#!/usr/bin/env python3
"""
Example usage of the enhanced Grazie client with LLM parameters support.
"""

from grazie_client import GrazieClient

def main():
    # Initialize client
    client = GrazieClient(environment="staging")
    
    print("Available models:")
    for model in client.get_available_models()[:5]:  # Show first 5
        print(f"  - {model}")
    
    print("\n" + "="*50)
    
    # Example 1: Simple chat with temperature parameter
    print("Example 1: Creative response with high temperature")
    creative_params = client.create_creative_params("high")
    response = client.simple_chat(
        "Write a creative tagline for a space travel company",
        parameters=creative_params,
        prompt="creative-tagline-example"
    )
    print(f"Response: {response}")
    
    print("\n" + "="*50)
    
    # Example 2: Deterministic response
    print("Example 2: Deterministic response with seed")
    deterministic_params = client.create_deterministic_params(seed=12345)
    response = client.simple_chat(
        "Explain quantum computing in one sentence",
        parameters=deterministic_params,
        prompt="quantum-explanation"
    )
    print(f"Response: {response}")
    
    print("\n" + "="*50)
    
    # Example 3: JSON response format
    print("Example 3: Structured JSON response")
    json_params = client.create_json_response_params()
    response = client.simple_chat(
        "Return the latitude and longitude of Paris in JSON format",
        parameters=json_params,
        prompt="paris-coordinates"
    )
    print(f"Response: {response}")
    
    print("\n" + "="*50)
    
    # Example 4: Custom parameters
    print("Example 4: Custom parameters with length limit")
    custom_params = {
        'temperature': 0.7,
        'length': 50,  # Limit response to 50 tokens
        'top_p': 0.9
    }
    response = client.simple_chat(
        "Explain artificial intelligence",
        parameters=custom_params,
        prompt="ai-explanation-short"
    )
    print(f"Response: {response}")
    
    print("\n" + "="*50)
    
    # Example 5: Chat with metadata
    print("Example 5: Chat with response metadata")
    messages = [
        {"type": "system_message", "content": "You are a helpful assistant."},
        {"type": "user_message", "content": "What's the weather like?"}
    ]
    
    content, metadata = client.chat_stream_with_metadata(
        messages, 
        parameters={'temperature': 0.8},
        prompt="weather-query"
    )
    
    print(f"Content: {content}")
    print(f"Metadata: {metadata}")
    
    print("\n" + "="*50)
    
    # Example 6: Streaming with parameters
    print("Example 6: Streaming response with custom parameters")
    stream_params = {
        'temperature': 1.2,
        'top_k': 50,
        'stop_token': '.'  # Stop at first period
    }
    
    messages = [
        {"type": "user_message", "content": "Tell me about machine learning"}
    ]
    
    print("Streaming response:")
    for chunk in client.chat_stream(messages, parameters=stream_params, prompt="ml-explanation"):
        if chunk.get("type") == "Content":
            print(chunk.get("content", ""), end="", flush=True)
        elif chunk.get("type") == "FinishMetadata":
            print(f"\n[Finished: {chunk.get('reason')}]")
        elif chunk.get("type") == "QuotaMetadata":
            spent = chunk.get('spent', {}).get('amount', 'N/A')
            print(f"[Tokens spent: {spent}]")

if __name__ == "__main__":
    main() 

File: grazie-service/example_usage.js
#!/usr/bin/env node

/**
 * Example usage of the JavaScript Grazie client with async/await
 * Demonstrates various features including streaming, parameters, and metadata
 */

const GrazieClient = require('./grazie_client.js');

async function main() {
    try {
        // Initialize client
        const client = new GrazieClient(null, "staging");
        
        console.log("Available models:");
        const models = await client.getAvailableModels();
        for (const model of models.slice(0, 5)) {  // Show first 5
            console.log(`  - ${model}`);
        }
        
        console.log("\n" + "=".repeat(50));
        
        // Example 1: Simple chat with temperature parameter
        console.log("Example 1: Creative response with high temperature");
        const creativeParams = client.createCreativeParams("high");
        const response1 = await client.simpleChat(
            "Write a creative tagline for a space travel company",
            null,
            "openai-gpt-4o",
            creativeParams,
            "creative-tagline-example"
        );
        console.log(`Response: ${response1}`);
        
        console.log("\n" + "=".repeat(50));
        
        // Example 2: Deterministic response
        console.log("Example 2: Deterministic response with seed");
        const deterministicParams = client.createDeterministicParams(12345);
        const response2 = await client.simpleChat(
            "Explain quantum computing in one sentence",
            null,
            "openai-gpt-4o",
            deterministicParams,
            "quantum-explanation"
        );
        console.log(`Response: ${response2}`);
        
        console.log("\n" + "=".repeat(50));
        
        // Example 3: JSON response format
        console.log("Example 3: Structured JSON response");
        const jsonParams = client.createJsonResponseParams();
        const response3 = await client.simpleChat(
            "Return the latitude and longitude of Paris in JSON format",
            null,
            "openai-gpt-4o",
            jsonParams,
            "paris-coordinates"
        );
        console.log(`Response: ${response3}`);
        
        console.log("\n" + "=".repeat(50));
        
        // Example 4: Custom parameters
        console.log("Example 4: Custom parameters with length limit");
        const customParams = {
            temperature: 0.7,
            length: 50,  // Limit response to 50 tokens
            top_p: 0.9
        };
        const response4 = await client.simpleChat(
            "Explain artificial intelligence",
            null,
            "openai-gpt-4o",
            customParams,
            "ai-explanation-short"
        );
        console.log(`Response: ${response4}`);
        
        console.log("\n" + "=".repeat(50));
        
        // Example 5: Chat with metadata
        console.log("Example 5: Chat with response metadata");
        const messages = [
            { type: "system", content: "You are a helpful assistant." },
            { type: "user", content: "What's the weather like?" }
        ];
        
        const [content, metadata] = await client.chatStreamWithMetadata(
            messages,
            "openai-gpt-4o",
            { temperature: 0.8 },
            "weather-query"
        );
        
        console.log(`Content: ${content}`);
        console.log(`Metadata: ${JSON.stringify(metadata, null, 2)}`);
        
        console.log("\n" + "=".repeat(50));
        
        // Example 6: Streaming with parameters
        console.log("Example 6: Streaming response with custom parameters");
        const streamParams = {
            temperature: 1.2,
            top_k: 50,
            stop_token: '.'  // Stop at first period
        };
        
        const streamMessages = [
            { type: "user", content: "Tell me about machine learning" }
        ];
        
        console.log("Streaming response:");
        for await (const chunk of client.chatStream(streamMessages, "openai-gpt-4o", streamParams, "ml-explanation")) {
            if (chunk.type === "Content") {
                process.stdout.write(chunk.content || "");
            } else if (chunk.type === "FinishMetadata") {
                console.log(`\n[Finished: ${chunk.reason}]`);
            } else if (chunk.type === "QuotaMetadata") {
                const spent = chunk.spent?.amount || 'N/A';
                console.log(`[Tokens spent: ${spent}]`);
            }
        }
        
        console.log("\n" + "=".repeat(50));
        
        // Example 7: Using different models (if available)
        console.log("Example 7: Trying different models");
        const availableModels = await client.getAvailableModels();
        const preferredModels = ["openai-gpt-4o", "anthropic-claude-3-5-sonnet-20241022"];
        
        for (const model of preferredModels) {
            if (availableModels.includes(model)) {
                console.log(`\nTesting with ${model}:`);
                try {
                    const response = await client.simpleChat(
                        "Hello, introduce yourself briefly",
                        null,
                        model,
                        { temperature: 0.5 }
                    );
                    console.log(`${model}: ${response.slice(0, 100)}...`);
                } catch (error) {
                    console.log(`${model}: Error - ${error.message}`);
                }
            }
        }
        
    } catch (error) {
        console.error("Error:", error.message);
        console.error("Make sure you have set the GRAZIE_JWT_TOKEN or USER_JWT_TOKEN environment variable");
    }
}

// Run the main function
if (require.main === module) {
    main().catch(console.error);
}

module.exports = { main }; 